OpenRA.Mods.Common/Traits/AutoCarryall.cs

```diff
@@ -92,45 +147,67 @@ void FindCarryableForTransport(Actor self)
 
 			foreach (var p in carryables)
 			{
+				UnderAutoCommand = true;
+
 				// Check if its actually me who's the best candidate
 				if (IsBestAutoCarryallForCargo(self, p.Actor) && ReserveCarryable(self, p.Actor))
 				{
-					busy = true;
 					self.QueueActivity(false, new FerryUnit(self, p.Actor));
 					break;
 				}
 			}
 		}
 
+		protected override void ResolveOrder(Actor self, Order order)
+		{
+			UnderAutoCommand = false;
+			base.ResolveOrder(self, order);
+		}
+
 		sealed class FerryUnit : Activity
 		{
 			readonly Actor cargo;
-			readonly Carryable carryable;
-			readonly Carryall carryall;
+			readonly AutoCarryable carryable;
+			readonly AutoCarryall carryall;
 
 			public FerryUnit(Actor self, Actor cargo)
 			{
 				this.cargo = cargo;
-				carryable = cargo.Trait<Carryable>();
-				carryall = self.Trait<Carryall>();
+				carryable = cargo.Trait<AutoCarryable>();
+				carryall = self.Trait<AutoCarryall>();
 			}
 
 			protected override void OnFirstRun(Actor self)
 			{
-				if (!cargo.IsDead && !carryall.IsTraitDisabled)
+				if (carryall.IsTraitDisabled || carryall.Carryable == null || carryall.Carryable.IsDead || !carryall.UnderAutoCommand)
+					Cancel(self, true);
+				else
 					QueueChild(new PickupUnit(self, cargo, 0, carryall.Info.TargetLineColor));
 			}
 
+			public override IEnumerable<TargetLineNode> TargetLineNodes(Actor self)
+			{
+				if (ChildActivity != null)
+					foreach (var n in ChildActivity.TargetLineNodes(self))
+						yield return n;
```

Comment:
```suggestion
				if (ChildActivity != null)
				{
					if (ChildActivity is PickupUnit)
					{
						yield return new TargetLineNode(Target.FromActor(cargo), carryall.Info.TargetLineColor);
						if (carryable.Destination != null)
							yield return new TargetLineNode(Target.FromCell(self.World, carryable.Destination.Value), carryall.Info.TargetLineColor);
					}
					else
						foreach (var n in ChildActivity.TargetLineNodes(self))
							yield return n;
				}
```

Can fix the target line issue like this

Reply:
Fixed